parsl/usage_tracking/usage.py

```diff
@@ -110,28 +111,70 @@ def __init__(self, dfk, port=50077,
         self.python_version = "{}.{}.{}".format(sys.version_info.major,
                                                 sys.version_info.minor,
                                                 sys.version_info.micro)
-        self.tracking_enabled = self.check_tracking_enabled()
-        logger.debug("Tracking status: {}".format(self.tracking_enabled))
+        self.tracking_level = self.check_tracking_level()
+        self.start_time = None
+        logger.debug("Tracking level: {}".format(self.tracking_level))
+
+    def check_tracking_level(self) -> int:
+        """Check if tracking is enabled and return level.
+
+        Checks the following:
+            1. PARSL_TRACKING environment variable
+                - Possible values:
+                    ["true", "false", "True", "False", "0", "1", "2", "3"]
+                - Other values are treated as Level 0 (disabled)
+
+            2. usage_tracking in Config
+                - Possible values: [True, False, 0, 1, 2, 3]
+
+            Parsl will choose the lowest level that is specified by any of the parameters, and 0 if none of them are set
+
+            True/False values are treated as Level 1/Level 0 respectively.
+
+        Returns: int
+            - 0 : Tracking is disabled
+            - 1 : Tracking is enabled with level 1
+                  Share info about Parsl version, Python version, platform
+            - 2 : Tracking is enabled with level 2
+                  Share info about config + level 1
+            - 3 : Tracking is enabled with level 3
+                  Share info about app count, app fails, execution time + level 2
+        """
+        inf = sys.maxsize
 
-    def check_tracking_enabled(self):
-        """Check if tracking is enabled.
+        envvar = str(os.environ.get("PARSL_TRACKING", None)).lower()
+        if envvar == "none":
+            envvar_level = inf
 
-        Tracking will be enabled unless either of these is true:
+        elif envvar == "false":
+            envvar_level = 0
 
-            1. dfk.config.usage_tracking is set to False
-            2. Environment variable PARSL_TRACKING is set to false (case insensitive)
+        elif envvar == "true":
+            envvar_level = 1
 
-        """
-        track = True
+        elif envvar in {"0", "1", "2", "3"}:
+            envvar_level = int(envvar)
+
+        else:
+            raise ConfigurationError(
+                f"PARSL_TRACKING values must be true, false, 0, 1, 2, or 3 and not {os.environ.get('PARSL_TRACKING')}"
+            )
+
+        if self.config.usage_tracking is None:
+            config_level = inf
+
+        elif 0 <= int(self.config.usage_tracking) <= 3:
+            config_level = int(self.config.usage_tracking)
```

Comment:
Thanks to `typeguard` this is a useless conversion.  `self.config.usage_tracking`, if not None, is guaranteed to be an integer.  Further, if the `usage_tracking` is always set (as discussed in another comment), the previous branch (`is None`) is also unnecessary.